imports/plugins/included/jobcontrol/server/jobs/cart.js

```diff
@@ -0,0 +1,97 @@
+import later from "later";
+import moment from "moment";
+import { Accounts, Cart, Jobs } from "/lib/collections";
+import { Hooks, Logger, Reaction } from "/server/api";
+import { ServerSessions } from "/server/publications/collections/sessions";
+
+
+Hooks.Events.add("onJobServerStart", () => {
+  Logger.debug("Adding Job removeStaleCart and Accounts to jobControl");
+  const settings = Reaction.getShopSettings();
+  if (settings.cart) {
+    new Job(Jobs, "cart/removeFromCart", {})
+      .priority("normal")
+      .retry({
+        retries: 5,
+        wait: 60000,
+        backoff: "exponential" // delay by twice as long for each subsequent retry
+      })
+      .repeat({
+        schedule: later.parse.text("every day")
+      })
+      .save({
+        cancelRepeats: true
+      });
+  } else {
+    Logger.warn("No cart cleanup schedule");
+  }
+});
+
+export default () => {
+  const removeStaleCart = Jobs.processJobs("cart/removeFromCart", {
+    pollInterval: 60 * 60 * 1000, // backup polling, see observer below
+    workTimeout: 180 * 1000
+  }, (job, callback) => {
+    Logger.debug("Processing cart/removeFromCart");
+    const settings = Reaction.getShopSettings();
+    if (settings.cart) {
+      const schedule = (settings.cart.cleanupDurationDays).match(/\d/);// configurable in shop settings
+      const olderThan = moment().subtract(Number(schedule[0]), "days")._d;
+      const cartDetails = Cart.find({
+        updatedAt: {
+          $lte: olderThan
+        }
+      }).fetch();
+      cartDetails.map(cart => {
+        const anonUser = Accounts.findOne({
+          _id: cart.userId,
+          emails: []
+        });
+        if (cart.items) {
+          if (anonUser) {
+            const removeCart = Cart.remove({ userId: anonUser._id });
+            const removeAccount = Accounts.remove(
+              {
+                _id: cart.userId,
+                emails: []
+              }
+            );
+            const destroySession = ServerSessions.remove({ _id: cart.sessionId });
```

Comment:
This logic is a little hard to follow and this is hard to test (not your fault, it's just the nature of these jobs).

Can we break this out into a function that could be called at the command line e.g. `purgeAnonymousCarts(3)` that would purge Anonymous carts older than 3 days (or whatever argument is passed there)? Even better if we could specify time unit, e.g. "1 day" or "2 hours". Then I would feel better about being able to test that it does what it says it's doing as I could just call it from the meteor shell.